frontend/src/screens/achievements.tsx

```diff
@@ -0,0 +1,197 @@
+                return 'Continue'
+            }
+
+            return 'Start'
+        })
+    }, [completedStudies, isComplete])
+
+    const convertBase64ToPdf = (base64PDF: string) => {
+        const byteCharacters = atob(base64PDF);
+        const byteNumbers = new Array(byteCharacters.length);
+        for (let i = 0; i < byteCharacters.length; i++) {
+            byteNumbers[i] = byteCharacters.charCodeAt(i);
+        }
+        const byteArray = new Uint8Array(byteNumbers);
+        const blob = new Blob([byteArray], { type: 'application/pdf' });
+        const url = URL.createObjectURL(blob);
+        return url;
+    };
+
+    const handleButtonClick = async () => {
+        if(!isComplete){
+            const incompleteStudy = studies.find((study) => !study.completedAt)
+            if(incompleteStudy){
+                launchStudy(api, incompleteStudy.id)
+            }
+        }else{
+            try{
+                const email = user.contactInfos?.find(e => e.type == 'EmailAddress')?.value
+                if(!email){
+                    setError(true)
+                    return
+                }
+
+                const response = await api.getBadgeCertificate({
+                    badgeId: studies[0].learningPath?.badgeId || '',
+                    email: email,
+                });
+                const pdfUrl = convertBase64ToPdf(response.pdf || '');
+                window.open(pdfUrl, '_blank');
+            }catch(error){
+                setError(true)
+            }
+        }       
+    }
+
+    return(
+        <Stack>
+            {error? <Alert title="Error" color="red" withCloseButton onClose={() => setError(false)}>Fail to download the PDF. Please try again later.</Alert> : ''}
+            <Modal c={colors.text} opened={opened} onClose={close} title={<Text size="lg" pl={20} pt={10} fw={700}>{learningPath}</Text>} centered>
+                <Stack pl={20} pr={20} pb={20}>
+                    <Stack>
+                        <Text size="sm">
+                            <span>{studies[0].learningPath?.badge?.description}</span><br/>
+                            <div style={{ marginTop: '.5rem' }}>{studies[0].learningPath?.badge?.tags?.map((tag) => {
+                                return '#'+tag+' ' 
+                            })}</div>
+                        </Text>
+                    </Stack>
+                    <Stack>
+                        {criteria.map((cr) => {
+                            return (
+                                <Group key={cr.title}>
+                                    <Text size="sm"><span style={{ fontWeight: '700' }}>{cr.title}</span>{cr.text}</Text>
+                                </Group>
+                            )
+                        })}
+                    </Stack>
+                    <Group>
+                        <Button onClick={() => handleButtonClick()} variant='filled' color={colors.purple}>{btnText}</Button>
```

Comment:
Maybe you can extract this out to it's own component so this component gets cleaned up a little with the above logic